Library/Homebrew/cmd/test-bot.rb

```diff
@@ -520,7 +521,7 @@ def formula formula_name
               test "brew", "uninstall", "--force", *unchanged_build_dependencies
               unchanged_dependencies -= unchanged_build_dependencies
             end
-            test "brew", "install", bottle_filename
+            test "brew", "install", "--force-bottle", bottle_filename
```

Comment:
I think it'd be unintuitive for `brew install` to install from source while someone's fiddling with their formula, then revert to using the bottle when they commit their changes.

But if we can't preserve that behavior without all these problems it introduced, then it can be changed to just look for uncommitted changes.